include/athena/core/context/internal/Edge.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 Athena. All rights reserved.
+ * https://getathena.ml
+ *
+ * Licensed under MIT license.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#ifndef ATHENA_EDGE_H
+#define ATHENA_EDGE_H
```

Comment:
I think, it may cause you tons of problems, if you have Edge.h somewhere else.

```suggestion
#pragma once
```

Reply:
Sorry, later